skrub/datasets/_fetching.py

```diff
@@ -371,6 +371,10 @@ def _fetch_figshare(
             UserWarning,
             stacklevel=2,
         )
+        import_optional_dependency(
+            "pyarrow", extra="pyarrow is required for parquet support."
+        )
```

Comment:
I think that I'd prefer to implement the try-except ourselves rather than import from a private pandas module.